protostar/self/cache_io.py

```diff
@@ -0,0 +1,32 @@
+from typing import Optional
+import json
+from pathlib import Path
+
+
+class CacheIO:
+    _CACHE_DIR_NAME = ".protostar_cache"
+    _EXTENSION = ".json"
+
+    def __init__(self, project_root_path: Path):
+        self._cache_path = project_root_path / Path(self._CACHE_DIR_NAME)
+        self._cache_path.mkdir(exist_ok=True)
+
+    def write(self, name: str, value: dict, override=True) -> None:
+        if not override and self._cache_path.exists():
+            return
+
+        with open(
+            self._cache_path / (name + self._EXTENSION), "w", encoding="utf-8"
+        ) as file:
+            file.write(json.dumps(value))
```

Comment:
I don't see why we would do that. The JSON file that's being created is not meant to be read by anyone but this software and it doesn't need formatting to understand it. Formatting would only add additional, redundant operations. I think formatting only makes sense when it may be viewed by humans.